TODO/the-ultimate-guide-to-creating-a-mobile-application.md

```diff
@@ -8,191 +9,191 @@
 
-- setting the task and initial scope of works
-- estimation
-- user/market research
-- UX wireframing
-- prototyping
-- UI design
-- animation
-- software architecture planning
-- iOS development
-- testing
-- release
-- updates.
+- 设立任务和初始工作范围
+- 估算
+- 用户/市场调研
+- UX 线框图
+- 原型
+- UI 设计
+- 动画
+- 软件架构设计
+- iOS 开发
+- 测试
+- 打包
+- 更新。
 
-Although you see the sequence of stages, it doesn’t mean that every further phase starts only when the previous one is completed. It’s impossible to imagine this kind of linear dependence as many processes and stages are interconnected even not being consequent in the presented list. Moreover, some of them like testing or estimation come up here and there being spread for all the process of app creation. Now let’s go along this way step by step to see how a subtle idea becomes a real mobile app.
+尽管上面的排列是有序的，但这并不意味着你需要在完成前者之后才开始下一步。尽管上列的众多过程和阶段是有联系的，但也不能简单臆测成线性关系。不仅如此，其中的一些如测试或估算会贯穿整个流程。现在就让我们来一步一步地将灵光一闪变成真正的应用。
 
 ![](https://cdn-images-1.medium.com/max/800/0*sUp1d1u14doiNvBH.png)
 
 ### Setting the task and initial scope of works
 
-As we have already mentioned in the article considering [logo design stages](http://tubikstudio.com/logo-design-creative-stages/), the point of setting the task is the foundation of all the design and then development process. At this point the team of designers and developers should obtain maximum information from the client to mark the right way to the goal. The one who walks without a destination in mind, will possibly come nowhere. In product design it works the same way: to get the result, you should clearly set the goals from the very start of the path. It doesn’t mean that the goals should stay totally the same at the end of the journey: the proper level of flexibility has to be set as the objectives can modify more or less in the creative process, research and testing. Still, if the general goals are not set at the start, the creative process can get easily transformed in a mess.
+就像我们之前在 [logo design stages](http://tubikstudio.com/logo-design-creative-stages/) 中提过的那样，设立任务这一点是设计和随后的开发的基础。在这个阶段设计师和开发者应该从客户那里获取尽可能多的信息来设立自己抵达目的地的路标。若行走无向，则寸步难进。产品设计也是一样的：若想有所产出，则必须在开始前就明确目标。这并不意味着在整个过程中目标是一成不变的：当客观事物发生变化时，目标也要随之灵活调整，重新调研和测试。话虽如此，但如果在开始时没有设立目标的话，整个过程就会轻易变成一团乱麻。
 
-Another vital tip, which we have seen through long practice, is that communicating with clients you should get not only their wishes but also try to get the ideas and reasons behind these wishes. If you understand why your customer wants to see particular colors, shapes or transitions, it will be easier for you (if necessary) to justify other methods of realization for these ideas which would give the result desired by the customer and at the same time user-friendly and thoughtfully consuming resources or enabling interactions.
+我们在长期实践中得出的另一个关键点是和用户交流时不仅要了解他们的需求，也要努力明白需求背后的想法和理由。如果你明白为什么你的用户想要特定的颜色、形状或过渡后，（如果必要的话）就可以调整其他方面来实现这些想法，也可以达成用户的愿望，与此同时还可以提升用户体验、考虑资源消耗或提升用户体验。
```

Comment:
it will be easier for you (if necessary) to justify other methods of realization for these ideas which would give the result desired by the customer and at the same time user-friendly and thoughtfully consuming resources or enabling interactions.

（必要时）你就更容易在其它的地方将这些想法转化成相应的实现，以满足用户的需求，同时也能在保证用户体验的前提下，兼顾资源消耗及交互控制。 （ 这个 or enabling interactions 真是读不懂）